app/javascript/components/visualization/controls/DifferentialExpressionGroupPicker.jsx

```diff
@@ -47,90 +49,94 @@ function parseDeFile(tsvText) {
  *
  * @return {Array} deGenes Array of DE gene objects, each with properties:
  *   name: Gene name
  *   score: Differential expression score assigned by Scanpy.
  *   log2FoldChange: Log-2 fold change.  How many times more expression (1 = 2, 2 = 4, 3 = 8).
  *   pval: p-value.  Statistical significance of the `score` value.
- *   pvalAdj: Adjusted p-value.  p-value adjusted for false discovery rate (FDR).
+ *   pvalAdj: Adjusted p-value.  p-value adjusted with Benjamini-Hochberg FDR correction
  *   pctNzGroup: Percent non-zero, group.  % of cells with non-zero expression in selected group.
  *   pctNzReference: Percent non-zero, reference.  % of cells with non-zero expression in non-selected groups.
  **/
-async function fetchDeGenes(bucketId, deFileName, numGenes=20) {
-  const deFilePath = `_scp-internal/differential-expression/${deFileName}`.replaceAll('/', '%2F')
-
-  // TODO (SCP-4321): Perhaps refine logic for fetching file from bucket, e.g. perhaps add
-  //  token parameter to fetchFileFromBucket
+async function fetchDeGenes(bucketId, deFilePath, numGenes=15) {
   const data = await fetchBucketFile(bucketId, deFilePath)
   const tsvText = await data.text()
   const deGenes = parseDeFile(tsvText)
-
   return deGenes.slice(0, numGenes)
 }
 
 /** Pick groups of cells for differential expression (DE) */
 export default function DeGroupPicker({
-  exploreInfo, setShowDeGroupPicker, setDeGroup, setDeGenes
+  bucketId, clusterName, annotation, deGenes, deGroup, setDeGroup, setDeGenes, setDeFileUrl,
+  countsByLabel
 }) {
-  const annotation = exploreInfo?.annotationList?.default_annotation
-  const groups = annotation?.values ?? []
+  const groups = getLegendSortedLabels(countsByLabel)
 
-  const [group, setGroup] = useState(noneSelected)
+  /** Update group in differential expression picker */
+  async function updateDeGroup(newGroup) {
+    setDeGroup(newGroup)
 
-  /** Update group in DE picker */
-  async function updateDeGroup() {
-    const bucketId = exploreInfo?.bucketId
-
-    // TODO (SCP-4321): Incorporate any updates to this general file name structure
-    // <cluster_name>--<annotation_name>--<group_name>--<annotation_scope>--<method>.tsv
     const deFileName = `${[
-      exploreInfo?.annotationList?.default_cluster,
+      clusterName,
       annotation.name,
-      group,
+      newGroup,
+      annotation.scope,
       'wilcoxon'
     ]
       .map(s => s.replaceAll(nonAlphaNumericRegex, '_'))
       .join('--') }.tsv`
 
-    const deGenes = await fetchDeGenes(bucketId, deFileName)
+    const basePath = '_scp_internal/differential_expression/'
+    const deFilePath = `${basePath}${deFileName}`.replaceAll('/', '%2F')
+
+    const deGenes = await fetchDeGenes(bucketId, deFilePath)
 
-    setDeGroup(group)
+    setDeGroup(newGroup)
     setDeGenes(deGenes)
 
-    setShowDeGroupPicker(false)
+    const baseUrl = 'https://storage.googleapis.com/download/storage/v1/'
```

Comment:
Non-blocking: We do something similar to this in scp-api.jsx `fetchBucketFile` - I feel like this could be abstracted into a helper function and referenced in both places.  In case Google ever updates/changes storage URLs on us.

Reply:
Good point, I've added this as an AC in SCP-4375.